app/modules/SecurityModule.scala

```diff
@@ -0,0 +1,67 @@
+package modules
+
+import com.google.inject.{AbstractModule, Provides}
+import net.codingwell.scalaguice.ScalaModule
+import org.pac4j.core.client.Clients
+import org.pac4j.core.config.Config
+import org.pac4j.oidc.client.OidcClient
+import org.pac4j.oidc.config.{KeycloakOidcConfiguration, OidcConfiguration}
+import org.pac4j.oidc.profile.OidcProfile
+import org.pac4j.core.profile.CommonProfile
+import org.pac4j.play.scala.{DefaultSecurityComponents, Pac4jScalaTemplateHelper, SecurityComponents}
+import org.pac4j.play.{CallbackController, LogoutController}
+import org.pac4j.play.store.{PlayCacheSessionStore, PlayCookieSessionStore, PlaySessionStore, ShiroAesDataEncrypter}
+import play.api.{Configuration, Environment}
+import play.cache.SyncCacheApi
+
+
+
+
+class SecurityModule extends AbstractModule {
+  override def configure(): Unit = {
+
+    val sKey = System.getenv("PLAY_SECRET_KEY").substring(0, 16)
+//    val dataEncrypter = new ShiroAesDataEncrypter(sKey)
+
+//    new PlayCacheSessionStore(getProvider(SyncCacheApi.))
+//    val playSessionStore = new PlayCookieSessionStore(dataEncrypter)
+//    bind(classOf[PlaySessionStore]).toInstance(playSessionStore)
+    bind(classOf[PlaySessionStore]).to(classOf[PlayCacheSessionStore])
+    bind(classOf[SecurityComponents]).to(classOf[DefaultSecurityComponents])
+
+    bind(classOf[Pac4jScalaTemplateHelper[CommonProfile]])
+
+    // callback
+    val callbackController = new CallbackController()
+    callbackController.setDefaultUrl("/?defaulturlafterlogout")
+    callbackController.setMultiProfile(true)
+    callbackController.setRenewSession(false)
+    bind(classOf[CallbackController]).toInstance(callbackController)
+
+    // logout
+    val logoutController = new LogoutController()
+    logoutController.setDefaultUrl("/")
+    bind(classOf[LogoutController]).toInstance(logoutController)
+  }
+
+  @Provides
+  def provideOidcClient: OidcClient[OidcProfile, OidcConfiguration] = {
+
+    val oidcConfiguration = new OidcConfiguration()
+    oidcConfiguration.setClientId("tdr")
+    oidcConfiguration.setSecret("1c66e726-520f-4c26-a65b-5be292249af8")
```

Comment:
Just in case anyone else is worried about this - we've reset the secret in Keycloak.

The Keycloak instance wasn't actually being used for anything in the deployed environment, so you wouldn't be able to access anything with tokens generated using this secret.